commcare_connect/opportunity/forms.py

```diff
@@ -683,6 +692,8 @@ def clean(self):
                     "optional_deliver_units",
                     error=f"{deliver_unit_obj.name} cannot be marked both Required and Optional",
                 )
+            if cleaned_data["end_date"] < now().date():
```

Comment:
If it's optional it will throw error. I recently faced this issue on staging.
I am guessing this is the same error [Sentry error](https://dimagi.sentry.io/issues/6077952930/?environment=staging&project=4505635339829248&query=is%3Aunresolved%20issue.priority%3A%5Bhigh%2C%20medium%5D&referrer=issue-stream&sort=date&statsPeriod=90d&stream_index=5)

Reply:
Fixed. [fe18fc6](https://github.com/dimagi/commcare-connect/pull/421/commits/fe18fc6fd656d3f303f7bb4919a4461d4d68f9e2)